### PR TITLE
Support numbers within prose.

### PR DESCRIPTION
**Pull request by wolfmanstout**
Allows user to say "numb" followed by a number within prose and have it formatted in decimal.

### CODE REVIEW

File: modes/dictation_mode.talon [hunk 1]
```diff
@@ -65,6 +65,11 @@ formatted <user.format_text>:
 ^format selection <user.formatters>$:
     user.formatters_reformat_selection(formatters)
 
+# Numbers
+numb <user.number_string>: "{number_string}"
+numb <user.number_string> colon <user.number_string>: "{number_string_1}:{number_string_2}"
+numb <user.number_string> (dot | point) <digit_string>: "{number_string}.{digit_string}"
```
Comment (rntz) [inline modes/dictation_mode.talon:71]: 1. Any reason not just include `prose_number` inside `raw_prose`?
2. Failing that, any reason not to do `<user.prose_number>: "{prose_number}"`?

**Review by rntz — COMMENTED** [review]
Looks ok, although see comments. I use `numeral` for this. Any pros/cons of that vs `numb`? What does Dragon use by default?

File: modes/dictation_mode.talon (hunk 1 above)
Comment (wolfmanstout) [reply modes/dictation_mode.talon:71]: 1. Silly me! I see no reason not to.
2. I think it's helpful to support both numerals and numbers as words.

**Comment by wolfmanstout** [comment]
> Looks ok, although see comments. I use `numeral` for this. Any pros/cons of that vs `numb`? What does Dragon use by default?

Dragon uses numeral. We could certainly use that, perhaps as an alternative. Downside is two more syllables. I copied "numb" from pokey's fork. Considering "numb" will only match this when adjacent to a number I'm not worried about accidentally activating this capture.

**Comment by rntz** [comment]
"he was numb to the pain" might get misheard as "he was numb two the pain"

File: modes/dictation_mode.talon (hunk 1 above)
Comment (rntz) [reply modes/dictation_mode.talon:71]: great, if it gets included in `raw_prose` then these commands aren't needed (iiuc?) - they'll be handled by the `<user.raw_prose>: auto_insert(raw_prose)` command.

**Comment by wolfmanstout** [comment]
> "he was numb to the pain" might get misheard as "he was numb two the pain" --> "he was 2 the pain"

Fair enough I'm happy to switch to numeral. Will need to test with Dragon which sometimes behaves weirdly when you reuse its grammar.

File: modes/dictation_mode.talon (hunk 1 above)
Comment (wolfmanstout) [reply modes/dictation_mode.talon:71]: Yep. I'll make that change later.

**Comment by wolfmanstout** [comment]
Made the changes and tested with Dragon and Conformer. Pretty small change now!

**Review by rntz — APPROVED** [review]

**Pull request merged**